### PR TITLE
Save the GMFs in HDF5 format

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1467041.
The tests are running here: https://ci.openquake.org/job/zdevel_oq-risklib/751

### CODE REVIEW

File: openquake/commonlib/calculators/base.py [hunk 1]
```diff
@@ -461,13 +447,26 @@ def get_gmfs(calc):
...
-            gmfs = {k: expand(gmf, calc.sitecol)
-                    for k, gmf in calc.gmf_by_trt_gsim.iteritems()}
+        return read_gmfs_from_csv(calc)
+    # else from rupture
+    gmf = calc.datastore['/gmfs/col00'].value
+    if calc.datastore.parent:
+        haz_sitecol = calc.datastore.parent['sitecol']
+    else:
+        haz_sitecol = calc.sitecol
+    risk_indices = set(calc.sitecol.indices)  # N'' values
```
Comment (ptormene) [inline openquake/commonlib/calculators/base.py]: Please explain what's N''

File: openquake/commonlib/calculators/event_based.py [hunk 2]
```diff
@@ -66,12 +87,27 @@ def counts_per_rlz(num_sites, num_imts, rlzs_assoc, sescollection):
...
+    """
+    :param num_sites: the number of sites
+    :param num_imts: the number of IMTs
+    :param rlzs_assoc: an instance of RlzsAssoc
+    :param sescollection: a list of dictionaries tag -> SESRupture
+    :returns: the number of bytes required to store the GMFs
+    """
+    nbytes = 0
+    for sescol, gsims in zip(sescollection, rlzs_assoc.get_gsims_by_col()):
+        bytes_per_gmf = 4 + 8 * len(gsims) * num_imts
```
Comment (ptormene) [inline openquake/commonlib/calculators/event_based.py]: Please add a comment to describe this calculation

File: openquake/commonlib/calculators/event_based.py [hunk 3]
```diff
@@ -402,8 +452,8 @@ def compute_gmfs_and_curves(ses_ruptures, sitecol, rlzs_assoc, monitor):
     :param monitor:
         a Monitor instance
     :returns:
-        a dictionary trt_model_id -> curves_by_gsim
-        where the list of bounding boxes is empty
+        a dictionary (trt_model_id, gsim) -> haz_curves and
+        (trt_model_id, None) -> gmfs
```
Comment (ptormene) [inline openquake/commonlib/calculators/event_based.py]: Please fix the docstring

File: openquake/commonlib/datastore.py [hunk 4]
```diff
@@ -38,22 +31,52 @@ def __getattr__(self, name):
 DATADIR = os.environ.get('OQ_DATADIR', os.path.expanduser('~/oqdata'))
 
 
+def get_nbytes(dset):
+    """
+    Extract the size in bytes of a dataset, without dipping in the tree.
+    Returns None if the dataset is actually a group.
```
Comment (ptormene) [inline openquake/commonlib/datastore.py]: Please explain better

**Comment by ptormene** [comment]
Besides the minor comments, LGTM

File: openquake/commonlib/calculators/base.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/calculators/base.py]: Done

File: openquake/commonlib/calculators/event_based.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/calculators/event_based.py]: Done

File: openquake/commonlib/calculators/event_based.py (hunk 3 above)
Comment (micheles) [inline openquake/commonlib/calculators/event_based.py]: Done

File: openquake/commonlib/datastore.py (hunk 4 above)
Comment (micheles) [inline openquake/commonlib/datastore.py]: Done

**Pull request merged**